lib/src/model/library.dart

```diff
@@ -16,113 +17,55 @@ import 'package:dartdoc/src/model/model.dart';
-    super.visitClassElement(element);
-  }
-
-  @override
-  void visitConstructorElement(ConstructorElement element) {
-    packageGraph.populateModelNodeFor(element, resolvedLibrary);
-  }
-
-  @override
-  void visitEnumElement(EnumElement element) {
-    packageGraph.populateModelNodeFor(element, resolvedLibrary);
-    super.visitEnumElement(element);
-  }
-
-  @override
-  void visitExtensionElement(ExtensionElement element) {
-    packageGraph.populateModelNodeFor(element, resolvedLibrary);
-    super.visitExtensionElement(element);
-  }
-
-  @override
-  void visitFieldElement(FieldElement element) {
-    packageGraph.populateModelNodeFor(element, resolvedLibrary);
-  }
-
-  @override
-  void visitFieldFormalParameterElement(FieldFormalParameterElement element) {
-    packageGraph.populateModelNodeFor(element, resolvedLibrary);
-  }
-
-  @override
-  void visitFunctionElement(FunctionElement element) {
-    packageGraph.populateModelNodeFor(element, resolvedLibrary);
-  }
-
-  @override
-  void visitLibraryElement(LibraryElement element) {
-    packageGraph.populateModelNodeFor(element, resolvedLibrary);
-    super.visitLibraryElement(element);
-  }
-
-  @override
-  void visitMixinElement(MixinElement element) {
-    packageGraph.populateModelNodeFor(element, resolvedLibrary);
-    super.visitMixinElement(element);
-  }
-
-  @override
-  void visitMultiplyDefinedElement(MultiplyDefinedElement element) {
-    packageGraph.populateModelNodeFor(element, resolvedLibrary);
-    super.visitMultiplyDefinedElement(element);
-  }
-
-  @override
-  void visitMethodElement(MethodElement element) {
-    packageGraph.populateModelNodeFor(element, resolvedLibrary);
-  }
-
-  @override
-  void visitParameterElement(ParameterElement element) {
-    // [ParameterElement]s without names do not provide sufficiently distinct
-    // hashes / comparison, so just skip them all. (dart-lang/sdk#30146)
-  }
-
-  @override
-  void visitPrefixElement(PrefixElement element) {
-    packageGraph.populateModelNodeFor(element, resolvedLibrary);
-  }
-
-  @override
-  void visitPropertyAccessorElement(PropertyAccessorElement element) {
-    packageGraph.populateModelNodeFor(element, resolvedLibrary);
-  }
-
-  @override
-  void visitSuperFormalParameterElement(SuperFormalParameterElement element) {
-    packageGraph.populateModelNodeFor(element, resolvedLibrary);
-  }
-
-  @override
-  void visitTopLevelVariableElement(TopLevelVariableElement element) {
-    packageGraph.populateModelNodeFor(element, resolvedLibrary);
-  }
-
-  @override
-  void visitTypeAliasElement(TypeAliasElement element) {
-    packageGraph.populateModelNodeFor(element, resolvedLibrary);
-  }
-
-  @override
-  void visitTypeParameterElement(TypeParameterElement element) {
-    packageGraph.populateModelNodeFor(element, resolvedLibrary);
+  _ModelNodeGatherer(this.resolvedLibrary, this.packageGraph);
+
+  @override
+  void visitCompilationUnit(CompilationUnit node) {
```

Comment:
Why do we use visitor, if we just iterate manually?

Reply:
Well said! OK I've dropped the visitor, and made this a method on PackageGraph. Much simpler.